### ISSUE TITLE
[v.2.12.0] Release Tracker

### ISSUE THREAD

**Issue by atalman** [pytorch/pytorch#180506]
We cut a [release branch](https://github.com/pytorch/pytorch/tree/release/2.12) for the 2.12.0 release.

Our plan from this point from this point is roughly:

* Phase 1 (until 27/4/26): work on finalizing the release branch
* Phase 2 (after 27/4/26): perform extended integration/stability/performance testing based on Release Candidate builds.

This issue is for tracking cherry-picks to the release branch.

## Cherry-Pick Criteria

**Phase 1 (until 27/4/26):**

Only low-risk changes may be cherry-picked from main:

1. Fixes to regressions against the most recent minor release (e.g. 2.11.x for this release; see [module: regression issue list](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22module%3A+regression%22+))
2. Critical fixes for: [silent correctness](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+correctness+%28silent%29%22), [backwards compatibility](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+bc-breaking%22+), [crashes](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+crash%22+), [deadlocks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+deadlock%22+), (large) [memory leaks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+memory+usage%22+)
3. Critical fixes to new features introduced in the most recent minor release (e.g. 2.11.x for this release)
4. Test/CI fixes
5. Documentation improvements
6. Compilation fixes or ifdefs required for different versions of the compilers or third-party libraries
7. Release branch specific changes (e.g. change version identifiers)

Any other change requires special dispensation from the release managers (currently @atalman, @malfet ). If this applies to your change please write "Special Dispensation" in the "Criteria Category:" template below and explain.

**Phase 2 (after 27/4/26):**

Note that changes here require us to rebuild a Release Candidate and restart extended testing (likely delaying the release). Therefore, the only accepted changes are **Release-blocking** critical fixes for: [silent correctness](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+correctness+%28silent%29%22), [backwards compatibility](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+bc-breaking%22+), [crashes](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+crash%22+), [deadlocks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+deadlock%22+), (large) [memory leaks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+memory+usage%22+)

Changes will likely require a discussion with the larger release team over VC or Slack.

## Cherry-Pick Process

1. Ensure your PR has landed in master. This does not apply for release-branch specific changes (see Phase 1 criteria).
2. Create (but do not land) a PR against the [release branch](https://github.com/pytorch/pytorch/tree/release/2.12).
   <details>

    ```bash
    # Find the hash of the commit you want to cherry pick
    # (for example, abcdef12345)
    git log

    git fetch origin release/2.12
    git checkout release/2.12
    git cherry-pick -x abcdef12345

    # Submit a PR based against 'release/2.11' either:
    # via the GitHub UI
    git push my-fork

    # via the GitHub CLI
    gh pr create --base release/2.12
    ```
 
    You can also use the `@pytorchbot cherry-pick` command to cherry-pick your PR. To do this, just add a comment in your merged PR. For example:
   
    ```
    @pytorchbot cherry-pick --onto release/2.12 -c docs
    ```
    (`-c docs` - is the category of your changes - adjust accordingly):
    
    For more information, see [pytorchbot cherry-pick docs](https://github.com/pytorch/pytorch/wiki/Bot-commands#cherry-pick).

    </details>

  

3. Make a request below with the following format:

```
Link to landed trunk PR (if applicable):
* 

Link to release branch PR:
* 

Criteria Category:
* 
```

1. Someone from the release team will reply with approved / denied or ask for more information.
2. If approved, someone from the release team will merge your PR once the tests pass. **Do not land the release branch PR yourself.**

**NOTE: Our normal tools (ghstack / ghimport, etc.) do not work on the release branch.**

Please note HUD Link with branch CI status and link to the HUD to be provided here.
[HUD](https://hud.pytorch.org/hud/pytorch/pytorch/release%2F2.12)

cc @seemethere @malfet @pytorch/pytorch-dev-infra 

### Versions

2.12.0

**Comment by etaf** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/180441

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/180816

Criteria Category:
Test/CI fixes -  fix XPU CI test case failures.

----
@atalman merged

**Comment by mlazos** [comment]
Link to landed trunk PR (if applicable):

* https://github.com/pytorch/pytorch/pull/180497

Link to release branch PR:

* https://github.com/pytorch/pytorch/pull/180916

Criteria Category:
Critical - Fixes issue for VLLM streams x cudagraphs


----
@atalman merged

**Comment by pragupta** [comment]
Link to landed trunk PR (if applicable):
* DISABLED via github issue in trunk https://github.com/pytorch/pytorch/issues/180475

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/180927

Criteria Category:
Critical -  required for rocm trunk job since this is not getting disabled

----
@atalman merged

**Comment by chuanqi129** [comment]
Link to landed trunk PR (if applicable):

- N/A

Link to release branch PR:

- https://github.com/pytorch/pytorch/pull/180965

Criteria Category:
Critical

----
@atalman merged

**Comment by naromero77amd** [comment]
DO NOT MERGE THIS, @pragupta INITIATED INDIVIDUAL CHERRY-PICKS

Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/180277 (target fix)
* https://github.com/pytorch/pytorch/pull/179041 (prerequisite)
* https://github.com/pytorch/pytorch/pull/180526 (prerequisite)
* https://github.com/pytorch/pytorch/pull/177725 (prerequisite, re-land)
* https://github.com/pytorch/pytorch/pull/180554 (prerequisite)
* https://github.com/pytorch/pytorch/pull/180648 (follow-up, re-enables ROCm combo-kernel tests)

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/181014

Criteria Category:
Test/CI fixes

@atalman Please hold-off merging these while we have some additional discussions.

@atalman Closing per discussion on PR: https://github.com/pytorch/pytorch/pull/181014#issuecomment-4300018246 and slack.

**Comment by azahed98** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/176994

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/181441

Reverted 
* https://github.com/pytorch/pytorch/pull/182184

Criteria Category:
Fixes new feature - inductor materialization heuristic to unblock vllm

----

@yangw-dev merged

@atalman reverted caused failures on vllm CI/CD

**Comment by angelayi** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/181230

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/181454

Criteria Category:
Regression - https://github.com/pytorch/pytorch/issues/180912

---
@atalman merged

**Comment by janeyx99** [comment]
Links to landed PRs on trunk:
- https://github.com/pytorch/pytorch/pull/181161
- https://github.com/pytorch/pytorch/pull/181434

Link to release branch PR:
- https://github.com/pytorch/pytorch/pull/181608

Criteria Category: Special Dispensation


----
@malfet merged

**Comment by atalman** [comment]
Link to landed trunk PR (if applicable):
* NA

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/181658

Criteria Category:
Critical vllm Related 


---
@atalman merged

**Comment by atalman** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/181456

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/181661

Criteria Category:
Docs

----
@atalman merged

**Comment by Lucaskabela** [comment]
Link to landed trunk PR (if applicable):

* https://github.com/pytorch/pytorch/pull/181463

Link to release branch PR:

* https://github.com/pytorch/pytorch/pull/181751

Criteria Category:
Critical vllm Related


----
@atalman merged

**Comment by anijain2305** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/181873

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/181925

Criteria Category: regression - causing segfault both internall and externally. 3 issues - https://github.com/pytorch/pytorch/issues/180741, https://github.com/pytorch/pytorch/pull/178231 and https://fb.workplace.com/groups/1075192433118967/permalink/1939251640046371/,

----
@atalman merged

**Comment by williamwen42** [comment]
Link to landed trunk PR (if applicable):
- Not applicable

Link to release branch PR:
- https://github.com/pytorch/pytorch/pull/181940

Criteria Category:
Fix new feature - Fix crash introduced in new feature
Issue: https://github.com/pytorch/pytorch/issues/181817
Feature: https://github.com/pytorch/pytorch/pull/177809

----
@atalman wrote: This is Special Dispensation category . Merged

**Comment by atalman** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/181793

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/181795

Criteria Category:
Critical fix for https://github.com/pytorch/pytorch/issues/181563

----
@atalman merged

**Comment by weifengpy** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/182100

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/182157

Criteria Category:
Critical

**Comment by eellison** [comment]
Link to trunk pr :
- https://github.com/pytorch/pytorch/pull/182139

Link to release branch pr: 
- https://github.com/pytorch/pytorch/pull/182178


Criteria Category:
Critical

**Comment by malfet** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/182285

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/182311

Criteria Category:
Critical fix (silent correctness) — bool attention masks return incorrect results in the 1-pass MPS SDPA decode kernel.

----
@atalman merged

**Comment by malfet** [comment]
Cherry-pick PR for #181742 against `release/2.12`: #182372

- Category: Test/CI fixes (item 4)
- Risk: low — test-only changes (`common_methods_invocations.py`, `common_modules.py`, `linalg.py`); adjusts tolerances for macOS 26.3

----
@atalman wrote: Looks like this is still failing

**Comment by atalman** [comment]
Please note. We are in Phase 2 (after 27/4/26). We don't accept new cherry-picks at this point.

**Comment by atalman** [comment]
closing this release is delivered
